modules/nixos/main.py

```diff
@@ -23,316 +23,44 @@
     fallback=True,
 ).gettext
 
+def getCalamaresNixosModulePath():
+    return '/run/current-system/sw/lib/calamares/modules/nixos'
 
-# The following strings contain pieces of a nix-configuration file.
-# They are adapted from the default config generated from the nixos-generate-config command.
+def getCustomOrDefaultSnippet(snippetName):
+    configPath = getCalamaresNixosModulePath()
 
-cfghead = """# Edit this configuration file to define what should be installed on
-# your system.  Help is available in the configuration.nix(5) man page
-# and in the NixOS manual (accessible by running ‘nixos-help’).
+    # Check if a custom snippet exists for the config name
+    snippetConfigPath = '{}/defaultConfigs/snippets/{}.snippet'.format(configPath, snippetName)
+    customConfigFilePath = '{}/customConfigs/snippets/{}.snippet'.format(configPath, snippetName)
```

Comment:
I'm not a fan of having these two `defaultConfigs` and `customConfigs` things. I like the way systemd does this sort of thing better, where a file in `/usr/lib/` is generally overridden by a like file in `/run/` which is also overridden by a like file in `/etc/`. In Nix the `/usr/lib/` stuff is generally replaced with `/nix/store/...-foo/` for the same purpose.

So I think it would be better if the package just looked in its own output path, but allowed files in a directory like `/etc/calamares-nixos-extensions/` or something like that to override. Then you don't have to do this awkward dance of overriding the package to change stuff like this, and you can just put custom configs in `environment.etc` or something.